tsconfig.json

```diff
@@ -0,0 +1,24 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "target": "es6",
+        "noImplicitAny": true,
+        "moduleResolution": "node",
+        "sourceMap": true,
+        "outDir": "functions",
```

Comment:
Why do we choose to generate the whole functions dir rather than saying `outDir: .` and putting this in `functions`?